mjpc/tasks/allegro/allegro.cc

```diff
@@ -0,0 +1,101 @@
+#include "mjpc/tasks/allegro/allegro.h"
+#include <iostream>
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include <mujoco/mujoco.h>
+#include "mjpc/task.h"
+#include "mjpc/utilities.h"
+
+namespace mjpc {
+std::string Allegro::XmlPath() const {
+  return GetModelPath("allegro/task.xml");
+}
+std::string Allegro::Name() const { return "Allegro"; }
+
+// ------- Residuals for cube manipulation task ------
+//     Cube position: (3)
+//     Cube orientation: (3)
+//     Cube linear velocity: (3)
+//     Control:  u
+// ------------------------------------------
+void Allegro::ResidualFn::Residual(const mjModel* model, const mjData* data,
+                        double* residual) const {
+  int counter = 0;
+
+  // ---------- Cube position ----------
+  // TODO(vincekurtz): specify goal position in a better way
```

Comment:
the goal cube position should be the palm position. we can add this a site to the right_hand.xml and a framepos sensor to the task.xml. see the shadow hand example for reference